lib/src/front_end/ast_node_visitor.dart

```diff
@@ -618,7 +619,38 @@ class AstNodeVisitor extends ThrowingAstVisitor<void>
 
   @override
   void visitInstanceCreationExpression(InstanceCreationExpression node) {
-    throw UnimplementedError();
+    token(node.keyword, after: space);
+
+    // If there is an import prefix and/or constructor name, then allow
+    // splitting before the `.`. This doesn't look good, but is consistent with
+    // constructor calls that don't have `new` or `const`. We allow splitting
+    // in the latter because there is no way to distinguish syntactically
+    // between a named constructor call and any other kind of method call.
+    var calls = <Piece>[];
```

Comment:
[nit] I wouldn't name the type following an import prefix a "call". WDYT about `memberAccesses` or `memberReads`?

Reply:
Those are both a little too long and too specific for my taste. Eventually, I expect this code to get a decent amount of complexity, based on how much complexity there is in the prototype and the old style around method call chains. So I want a fairly terse and generic term for the things with a chain.

I changed it to "operation". I think that is suitably generic.